packages/core/src/web/app/components/settings/Path.tsx

```diff
@@ -1,98 +1,86 @@
 import React from 'react';
 
-import Controls from '@core/app/components/settings/Control';
-import onOffOptionFactory from '@core/app/components/settings/onOffOptionFactory';
-import SelectControl from '@core/app/components/settings/SelectControl';
-import type { WorkAreaModel } from '@core/app/constants/workarea-constants';
+import type { DefaultOptionType } from 'antd/es/select';
+
 import { getWorkarea } from '@core/app/constants/workarea-constants';
+import { useSettingStore } from '@core/app/pages/Settings/useSettingStore';
 import UnitInput from '@core/app/widgets/Unit-Input-v2';
 import i18n from '@core/helpers/i18n';
-import type { StorageKey } from '@core/interfaces/IStorage';
 
-interface Props {
-  defaultUnit: string;
-  getBeamboxPreferenceEditingValue: (key: string) => any;
-  loopCompensation: number;
-  selectedModel: WorkAreaModel;
-  updateBeamboxPreferenceChange: (item_key: string, newVal: any) => void;
-  updateConfigChange: (id: StorageKey, newVal: any) => void;
-}
+import SettingFormItem from './components/SettingFormItem';
+import SettingSelect from './components/SettingSelect';
 
-const Path = ({
-  defaultUnit,
-  getBeamboxPreferenceEditingValue,
-  loopCompensation,
-  selectedModel,
-  updateBeamboxPreferenceChange,
-  updateConfigChange,
-}: Props): React.JSX.Element => {
+const Path = (): React.JSX.Element => {
   const { lang } = i18n;
+  const { getConfig, getPreference, setConfig, setPreference } = useSettingStore();
+
+  const selectedModel = getPreference('model');
+  const defaultUnit = getConfig('default-units');
   const workarea = getWorkarea(selectedModel);
 
-  const vectorSpeedConstraintOptions = onOffOptionFactory(
-    getBeamboxPreferenceEditingValue('vector_speed_contraint') !== false,
-    { lang },
-  );
-  const precutSwitchOptions = onOffOptionFactory(getBeamboxPreferenceEditingValue('blade_precut'), {
-    lang,
-  });
+  const commonBooleanOptions = [
+    { label: lang.settings.on, value: true },
+    { label: lang.settings.off, value: false },
+  ] as unknown as DefaultOptionType[];
```

Comment:
why `Path` has its own `commonBooleanOptions` while others use prop from `index`?
BTW, do you think we need to move `commonBooleanOptions` into a seperate file if adding props is annoying?